### PR TITLE
[4.1][bugfix] Respect Core files

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue #36287 .

### Summary of Changes

- Overrides lookup need to be based on the template folder

### CODE REVIEW

**Comment by richard67** [comment]
Added release blocker label as inherited from the issue.

File: administrator/components/com_templates/src/Model/TemplateModel.php [hunk 1]
```diff
@@ -972,7 +972,7 @@ public function &getSource()
 				$item->source = file_get_contents($filePath);
 				$item->filePath = Path::clean($filePath);
 
-				if ($coreFile = $this->getCoreFile($fileName, $this->template->client_id))
+				if ($coreFile = $this->getCoreFile(str_replace(JPATH_ROOT . ($this->template->client_id === 1 ? '/administrator/' : '/') . 'templates/' . $this->template->element, '', $fileName), $this->template->client_id))
```
Comment (HLeithner) [inline administrator/components/com_templates/src/Model/TemplateModel.php]: can we get this more readable? maybe assign it to it's own variable?

Comment (dgrammatiko) [reply administrator/components/com_templates/src/Model/TemplateModel.php]: Is https://github.com/joomla/joomla-cms/pull/36290/commits/d01370e5e518598eb8d5e8e061bb5cf7342598a1 any better?

Comment (HLeithner) [reply administrator/components/com_templates/src/Model/TemplateModel.php]: yes thanks

**Comment by brianteeman** [comment]
I have tested this item :red_circle: unsuccessfully on d01370e5e518598eb8d5e8e061bb5cf7342598a1<br /><br />No change from reported bug<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36290">issues.joomla.org/tracker/joomla-cms/36290</a>.</sub>

**Comment by dgrammatiko** [comment]
> No change from reported bug

I'm guessing you're on a windows system...

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 5290f152d952b343598c64dbc6808e8eb0d5e7ec<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36290">issues.joomla.org/tracker/joomla-cms/36290</a>.</sub>

**Comment by jwaisner** [comment]
@dgrammatiko ,

  Can you please update this PR to be in sync with the latest branch?

**Comment by khu5h1** [comment]
I have tested this item :white_check_mark: successfully on 237b3544e143f4554d18062ac905f4948ab2502b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36290">issues.joomla.org/tracker/joomla-cms/36290</a>.</sub>

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 237b3544e143f4554d18062ac905f4948ab2502b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36290">issues.joomla.org/tracker/joomla-cms/36290</a>.</sub>

**Comment by jwaisner** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36290">issues.joomla.org/tracker/joomla-cms/36290</a>.</sub>

**Comment by bembelimen** [comment]
Thx

**Pull request merged**